starlette/testclient.py

```diff
@@ -449,16 +449,16 @@ def _choose_redirect_arg(
         self, follow_redirects: bool | None, allow_redirects: bool | None
     ) -> bool | httpx._client.UseClientDefault:
         redirect: bool | httpx._client.UseClientDefault = httpx._client.USE_CLIENT_DEFAULT
+        if allow_redirects is not None and follow_redirects is not None:
+            raise RuntimeError(  # pragma: no cover
+                "Cannot use both `allow_redirects` and `follow_redirects`."
+            )
         if allow_redirects is not None:
             message = "The `allow_redirects` argument is deprecated. Use `follow_redirects` instead."
             warnings.warn(message, DeprecationWarning)
```

Comment:
It seems that it was added [here](https://github.com/encode/starlette/commit/6765502c1e5a418c16bb379c54b8ea706b0444cf#diff-aca25e5f16c4fd49338ccdf3631f72455309335fee1e27f3d3b6016fa94ecedfR401).

I will work on the test, and if you decide that it should be removed, I will remove it.

Reply:
I saw that the `allow_redirects` was being covered in tests because it was being used in `test_responses.py` and in `tests/middleware/test_https_redirect.py`.

I changed it to `follow_redirects` as `allow_redirects` is deprecated and added tests to cover all the cases in `test_testclient`.